src/Nethermind/Nethermind.State/IStateProvider.cs

```diff
@@ -22,7 +21,7 @@ public interface IStateProvider : IReadOnlyStateProvider, IJournal<int>
 
         void CreateAccount(Address address, in UInt256 balance, in UInt256 nonce);
 
-        void UpdateCodeHash(Address address, Keccak codeHash, IReleaseSpec spec, bool isGenesis = false);
+        void InsertCode(Address address, ReadOnlyMemory<byte> code, IReleaseSpec spec, bool isGenesis = false);
```

Comment:
NP: Recommend Span instead of memory for the interface. You can make a Span from Memory, but not the other way around.